CodeReview.java

```diff
@@ -38,131 +38,151 @@ public int getLargestNumber() {
 
     public int expensiveFunctionExample() {
         int currNum = 0;
-        if (getLargestNumber() > currNum)
-            currNum = getLargestNumber();
-        for (int i=0; i < getLargestNumber(); ++i)
-            System.out.println(getLargestNumber());
+        int largestNum = getLargestNumber();
+
+        if (largestNum > currNum)
+            currNum = largestNum;
+
+        for (int i = 0; i < largestNum; ++i)
```

Comment:
Consider using braces. Use Foreach. Consider using logger instead of System.out.